src/plugins/mjpeg-video/supervisor/lib/Supervisor.js

```diff
@@ -0,0 +1,247 @@
+// Dependencies
+const Promise       = require( "bluebird" );
+const path		    = require( "path" );
+const execFileAsync = require( 'child-process-promise' ).execFile;
+const http          = require( "http" );
+const SIOServer     = require( "socket.io" );
+const ObjectPool    = require( "ObjectPool" );
+const Periodic      = require( "Periodic" );
+const EventEmitter  = require( "eventemitter3" );
+const fs 		    = Promise.promisifyAll( require( "fs" ) );
+
+// Logging utilities
+const log           = require( "debug" )( "app:supervisor:log" );
+const error		    = require( "debug" )( "app:supervisor:error" );
+
+const Camera        = require( "Camera" );
+const maxCameras    = 4;
+
+class Supervisor
+{
+    constructor( settings )
+    {
+        // Initialize properties
+        this.settings   = settings;
+        this.httpServer = null;
+        this.sioServer  = null;
+        this.eventBus   = new EventEmitter();
+        this.cameras    = {};
+        this.sslInfo    = 
+        {
+            certPath: this.settings.cert,
+            keyPath: this.settings.key
+        }
+
+        this.defaultCamSettings =
+        {
+            framerate: 30,
+            resolution: "1280x720"
+        }
+
+        // Create a pool of ports for the cameras to start daemons on
+        let wsPort          = 8200;
+        this.portPool       = new ObjectPool( { maxSize: maxCameras, initialSize: maxCameras }, () =>
+        {
+            // Create port and increment counter
+            return wsPort++;
```

Comment:
If the ports are intended to be discoverable, it would also be possible to use this to generate the port numbers which will grab whatever free port the OS has available. https://github.com/daaku/nodejs-freeport/blob/master/lib/freeport.js